tests/integration/pilot/validation_test.go

```diff
@@ -87,52 +81,55 @@ func TestValidation(t *testing.T) {
 			for _, cluster := range t.Clusters().Configs() {
 				for i := range dataset {
 					d := dataset[i]
+					for _, valid := range []bool{true, false} {
```

Comment:
Not sure if the 'invalid' case makes sense here, as the original purpose was to test invalid configurations. Now, it seems to test the annotation instead?

Reply:
Its not about testing specific invalid configs; it just needs *a* invalid config to verify the webhook actually rejects them. Now we have a much easier way to always trigger 'invalid' for these cases.

Reply:
Makes sense. I thought the intention was for testing specific invalid configs so I revised the logic

https://github.com/istio/istio/blob/4f017f95ba4efdf7af74ae3c2116eca7d2d7f10f/tests/integration/pilot/validation_test.go#L76-L84

to check `Required value` instead of webhook denial to make sure the new validator won't break existing validation logics.

BTW, doesn't TestWebhook already handle whether the webhook is rejecting things??

Reply:
It is slightly redundant, but this one also tests we test each crd + version. We have had it catch bugs before

Reply:
Got it